analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Assertions/ProgramStateAssertions.cs

```diff
@@ -44,9 +44,11 @@ public AndConstraint<ProgramStateAssertions> HaveUnknownException()
         return new AndConstraint<ProgramStateAssertions>(this);
     }
 
-    public AndConstraint<ProgramStateAssertions> HaveExceptionOfTypeException()
+    public AndConstraint<ProgramStateAssertions> HaveExceptionOfTypeException() => HaveExceptionOfType("Exception");
+
+    public AndConstraint<ProgramStateAssertions> HaveExceptionOfType(string exceptionTypeName)
     {
-        Execute.Assertion.ForCondition(Subject.Exception.ToString() == "Exception").FailWith("Expected program state to have an exception of type exception but fount {0}", Subject.Exception);
+        Execute.Assertion.ForCondition(Subject.Exception.ToString() == exceptionTypeName).FailWith($"Expected program state to have an exception of type {exceptionTypeName} but fount {{0}}", Subject.Exception);
```

Comment:
It looks weird to mix format string with interpolation. Let's use interpolation for both or formatting for both.
```suggestion
        Execute.Assertion.ForCondition(Subject.Exception.ToString() == exceptionTypeName).FailWith($"Expected program state to have an exception of type {exceptionTypeName} but found {Subject.Exception}");
```

Reply:
Typo should be solved by the previous PR